scripts/tests-functional/run.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -e
+
+divider="------------------------------------------------------------------------------------------"
+kind_cluster_name="loftsman-functional-tests"
+consul_chart_version="0.31.1"
+chartmuseum_container_name="${kind_cluster_name}-chartmuseum"
+
+this_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd $this_dir/../../
+
+if ! command -v helm &>/dev/null; then
+  echo "The helm binary needs to be installed to run functional tests"
+  exit 1
+fi
+if ! command -v kind &>/dev/null; then
+  echo "Kubernetes in Docker (KIND) is required to run functional tests (https://kind.sigs.k8s.io/)"
+  exit 1
+fi
+
+if kind get clusters | grep "^$kind_cluster_name$" &>/dev/null; then
+  kind delete cluster --name $kind_cluster_name
+fi
+rm $this_dir/charts/*.tgz &>/dev/null || true
+rm $this_dir/charts/*.yaml &>/dev/null || true
+docker rm -f $chartmuseum_container_name &>/dev/null || true
+
+revert_context=""
+if kubectl config current-context &>/dev/null; then
+  revert_context=$(kubectl config current-context)
+fi
+
+################################# Manifest chart sources ######################################
+echo $divider
+echo "Manifest chart sources"
+kind create cluster --name $kind_cluster_name
```

Comment:
It can take a bit on a cold system to have kind fully running after the exit (pods still coming up). May make sense to add some validation.

Reply:
added your review on the functional testing pieces specifically here: https://github.com/Cray-HPE/loftsman/issues/30. Definitely want to wait for a stable cluster state when we get this functional testing piece officially integrated. I was basically just trying to get something that was reasonable validation for me locally on the functional testing front considering the possible impact of changes in this effort.